source/config/featureFlag.py

```diff
@@ -0,0 +1,226 @@
+		optionsEnum: str,
+		behaviorOfDefault: str
+) -> FeatureFlag:
+	""" Used in conjunction with configObj.Validator
+	param value: The value to be validated / converted to a FeatureFlag object.
+	Expected: "enabled", "disabled", "default"
+	param behaviorOfDefault: Required, the default behavior of the flag, should be "enabled" or "disabled".
+	"""
+	log.debug(
+		f"Validating feature flag: {value}"
+		f", optionsEnum: {optionsEnum}"
+		f", behaviorOfDefault: {behaviorOfDefault}"
+	)
+	if not isinstance(optionsEnum, str):
+		raise ValidateError(
+			'Spec Error: optionsEnum must be specified as a string'
+			f" (got type {type(optionsEnum)} with value: {optionsEnum})"
+		)
+	try:
+		OptionsEnumClass = dict(featureFlagEnums.getAvailableEnums())[optionsEnum]
+	except KeyError:
+		raise ValidateError(
+			"Spec Error: optionsEnum must be an enum defined in the config.featureFlagEnums module."
+			f" (got {optionsEnum})"
+		)
+
+	if not isinstance(behaviorOfDefault, str):
+		raise ValidateError(
+			'Spec Error: behaviorOfDefault must be specified as a valid'
+			f' {OptionsEnumClass.__qualname__} member string'
+			f" (got type {type(behaviorOfDefault)} with value: {behaviorOfDefault})"
+		)
+	try:
+		behaviorOfDefault = OptionsEnumClass[behaviorOfDefault.upper()]
+	except KeyError:
+		raise ValidateError(
+			"Spec Error: behaviorOfDefault must be specified as a valid enum member string for enum class "
+			f"{OptionsEnumClass.__qualname__} (got {behaviorOfDefault})"
+		)
+	if behaviorOfDefault == OptionsEnumClass.DEFAULT:
+		raise ValidateError("Spec Error: behaviorOfDefault must not be 'default'/'DEFAULT'")
+
+	if not isinstance(value, str):
+		raise ValidateError(
+			'Expected a featureFlag value in the form of a string. EG "disabled", "enabled", or "default".'
+			f" Got {type(value)} with value: {value} instead."
+		)
+
+	try:
+		value = OptionsEnumClass[value.upper()]
+	except KeyError:
+		raise ValidateError(
+			"FeatureFlag value must be specified as a valid enum member string for enum class "
+			f"{OptionsEnumClass.__qualname__} (got {value})"
+		)
+
+	return FeatureFlag(value, behaviorOfDefault)
+
+
+def _transformSpec_AddFeatureFlagDefault(specString: str, **kwargs) -> str:
+	""" Ensure that default is specified for featureFlag used in configSpec.
+	Param examples based on the following spec string in configSpec:
+		loadChromiumVBufOnBusyState = featureFlag(behaviorOfDefault="enabled", optionsEnum="BoolFlag")
+	@param specString: EG 'featureFlag(behaviorOfDefault="enabled", optionsEnum="BoolFlag")'
+	@param kwargs: EG {'behaviorOfDefault': 'enabled', 'optionsEnum':'BoolFlag'}
+	@return 'featureFlag(behaviorOfDefault="ENABLED", optionsEnum="BoolFlag", default="DEFAULT")'
+	@remarks Manually specifying 'default' in the configSpec string (for featureFlag) will result in a
+		VdtParamError. Required params:
+		- 'behaviorOfDefault'
+		- 'optionsEnum'
+	"""
+	log.info(f"specString: {specString}, kwargs: {kwargs}")

```

Comment:
Why is this log present at info level? The logged message is not self-explaining.

If it is a debug log needed only for this PR's development, it should be removed. If it has any added value to for future debug, it may be kept at debug level and maybe with a more explicit message on its goal. If there is a justification to keep it at a so high level could you please elaborate?

@feerrenrut , @seanbudd, any idea?

Reply:
This should probably be debug level or removed

Reply:
Yes, I'll remove this. It was helpful for debugging during development.